src/cc/frontends/clang/b_frontend_action.cc

```diff
@@ -1826,6 +1824,20 @@ void BFrontendAction::DoMiscWorkAround() {
   else {
     probefunc = "";
   }
+  probefunc += "#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__\n"
+    "#define BCC_PROBE_READ(dst, src, probe_read_func) ({ \\\n"
+    "  int __sz = sizeof(*(dst)) < sizeof(*(src)) ? sizeof(*(dst)) : sizeof(*(src)); \\\n"
+    "  __builtin_memset((char *)(dst) + __sz, 0, sizeof(*(dst)) - __sz); \\\n"
+    "  probe_read_func(dst, sizeof(*(dst)), src); \\\n"
```

Comment:
Are lines 1828 and 1829 necessary?

Let us say *dst size is 4 and *src size is 8, __sz = 4, so effectively, the code is `probe_read_func(...)` as in the above.

Let us say *dst size is 8 and *src size is 4, __sz = 4. So we will have __builtin_memset((char *)dst + 4, 0, 4) which will set to 0 for last 4 bytes of *dst. But we will have probe_read_func(dst, 8, src) which will overwrite the last 4 bytes of *dst.

So I think we should just have probe_read_func(dst, sizeof(*(dst)), src).

Reply:
agree. Will do so.